Extensions/ClaimPrincipalExtension.cs

```diff
@@ -11,28 +11,94 @@ public static class ClaimsPrincipalExtension
         /// <param name="claimsPrincipal">Claims principal</param>
         /// <returns>A string corresponding to an account identifier as defined in <see cref="Microsoft.Identity.Client.AccountId.Identifier"/></returns>
         public static string GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
+        {
```

Comment:
NB
In single tenant scenarios, for a guest account, the ClaimsPrincipal does not contain enough information to compute the account id as the tenant id is the guest tenant (where the gest account was invited), and the oid is the guest oid. We'd need, for the Web scenario to also surface the guest AccountId